File: app/GOExercise/GOExercise/Models/WorkoutRecord.swift
import Foundation
import SwiftData

@Model
final class WorkoutRecord: Identifiable {
    @Attribute(.unique) var id: UUID
    var date: Date
    var categoryRaw: String
    var exercisesData: Data
    var memo: String?
    var createdAt: Date
    var updatedAt: Date

    var category: WorkoutCategory {
        get { WorkoutCategory(rawValue: categoryRaw) ?? .other }
        set {
            categoryRaw = newValue.rawValue
            updatedAt = Date()
        }
    }

    var exercises: [ExerciseItem] {
        get {
            (try? JSONDecoder().decode([ExerciseItem].self, from: exercisesData)) ?? []
        }
        set {
            exercisesData = (try? JSONEncoder().encode(newValue)) ?? Data()
            updatedAt = Date()
        }
    }

    init(
        id: UUID = UUID(),
        date: Date,
        category: WorkoutCategory,
        exercises: [ExerciseItem],
        memo: String? = nil,
        calendar: Calendar = .current,
        createdAt: Date = Date(),
        updatedAt: Date = Date()
    ) {
        self.id = id
        self.date = calendar.startOfDay(for: date)
        self.categoryRaw = category.rawValue
        self.exercisesData = (try? JSONEncoder().encode(exercises)) ?? Data()
        self.memo = memo
        self.createdAt = createdAt
        self.updatedAt = updatedAt
    }
}

File: app-android/app/src/main/java/com/goexercise/app/domain/WorkoutRecord.kt
package com.goexercise.app.domain

import java.time.LocalDate
import java.util.UUID

/**
 * 運動記録のドメインモデル(純粋ロジック用)。iOS `WorkoutRecord` の値部分の移植。
 *
 * 日付は **`LocalDate`(暦日)** で持つ。iOS は `Date` を `startOfDay` に丸めて保持するが、
 * ストリーク/達成判定は暦日単位なので Android は最初から LocalDate を正本にする
 * (タイムゾーン依存を排除し iOS と同入力同出力を担保)。Room 永続層では epochDay で保存する。
 */
data class WorkoutRecord(
    val id: String = UUID.randomUUID().toString(),
    val date: LocalDate,
    val category: WorkoutCategory,
    val exercises: List<ExerciseItem>,
    val memo: String? = null,
)
